### ISSUE TITLE
Folding decorations causes visual issues with the folded line

### ISSUE THREAD

**Issue by solemnify** [Microsoft/monaco-editor#751]
monaco-editor version: 0.10.1
Browser: Chrome
OS: Windows
Steps or JS usage snippet reproducing the issue:

Having whole line delta decorations with a transparent background color for indented lines causes the background colors to overlap and become unreadable when a parent line folds any children lines that contain these decorations.

**Comment by alexandrudima** [comment]
I'm not sure what you mean.

**Comment by solemnify** [comment]
@alexandrudima 

As an example, taking the example from "Rendering glyphs in the margin" via the playground.

I modified the content class to contain a transparent background:

```
.myContentClass {
	background: rgba(255, 0, 0, 0.25);
}
```

I modified the source to enable folding, added some additional indented code, and add the decorator to these new lines:

```
var jsCode = [
	'"use strict";',
	'function Person(age) {',
	'	if (age) {',
	'		this.age = age;',
	'		this.age = age;',
	'		this.age = age;',
	'		this.age = age;',
	'		this.age = age;',    
	'	}',
	'}',
	'Person.prototype.getAge = function () {',
	'	return this.age;',
	'};'
].join('\n');

var editor = monaco.editor.create(document.getElementById("container"), {
	value: jsCode,
    folding: true,
	language: "javascript",
	glyphMargin: true
});

var decorations = editor.deltaDecorations([], [
	{
		range: new monaco.Range(4,1,4,1),
		options: {
			isWholeLine: true,
			className: 'myContentClass',
		}
	},
	{
		range: new monaco.Range(5,1,5,1),
		options: {
			isWholeLine: true,
			className: 'myContentClass',
		}
	},
	{
		range: new monaco.Range(6,1,6,1),
		options: {
			isWholeLine: true,
			className: 'myContentClass',
		}
	},
	{
		range: new monaco.Range(7,1,7,1),
		options: {
			isWholeLine: true,
			className: 'myContentClass',
		}
	},
	{
		range: new monaco.Range(8,1,8,1),
		options: {
			isWholeLine: true,
			className: 'myContentClass',
		}
	},                
]);

// You can now use `decorations` to change or remove the decoration


```

Here's the before and after folding, as you can see after folding the background color of the folder is much more intense and it looks to be overlapping the background color from the children decorations.

Before:
![](https://i.imgur.com/mbDhLvu.png)

After:
![](https://i.imgur.com/5QNNIjN.png)

**Comment by alexandrudima** [comment]
@solemnify  :+1:

For a very long time, decorations on hidden lines would be rendered at the start of the folding location. I've changed this a few months ago once we started using an interval tree for storing decorations.

TL;DR: the behaviour you desire will come out with the next release (very soon).
